src/NodeApi/JSReference.cs

```diff
@@ -131,18 +142,26 @@ public void MakeStrong()
     /// Gets the referenced JS value, or null if the reference is weak and the value is
     /// no longer available.
     /// </summary>
+    /// <exception cref="ObjectDisposedException">The reference is disposed.</exception>
     public JSValue? GetValue()
     {
         JSValueScope.CurrentRuntime.GetReferenceValue(Env, _handle, out napi_value result)
             .ThrowIfFailed();
         return result;
     }
 
+    /// <summary>
+    /// Gets a value indicating whether the referenced JS value is still available (not released).
+    /// Returns false (does not throw) if the reference is disposed.
+    /// </summary>
+    public bool HasValue => !IsDisposed && (!IsWeak || GetValue().HasValue);
```

Comment:
I have seen many times how the `HasValue`-like property is misused in code and often causes issues. E.g. in the code developer may check `HasValue`, get a positive result, and act upon it, but the value is then destroyed in a different thread. A much better approach is to request a strong value and act on it depending if it is null or not.
Should we just add the `IsDisposed` check into the `GetValue()` method and use it instead?

Reply:
> check `HasValue`, get a positive result, and act upon it, but the value is then destroyed in a different thread

That isn't a concern with this API because references can only be accessed from a single thread.

I have also given feedback about mis-use of similar "check" APIs (such as `ContainsKey()` or `File.Exists()`) in many code reviews. But that doesn't invalidate the utility of such APIs in some legitimate use cases, as in the places where I'm using `HasValue` in this PR.

Without multithreading, there is still a minor issue of inefficiency if the API is mis-used. But that's no different from many other equally valid APIs.

> Should we just add the `IsDisposed` check into the `GetValue()` method and use it instead?

I think it is correct for `GetValue()` to throw `ObjectDisposedException` when the reference is disposed. That can help catch some bugs.

I don't have a problem with making the `HasValue` property `internal` if you prefer. But I thought it could be generally helpful in some valid use cases.